glossary.md

```diff
@@ -0,0 +1,241 @@
+
+The Merkle root is the root hash of a [Merkle Patricia tree] (MPT). A MPT is a
+[trie], which is a tree tree structure that maps keys to values. The root hash
+of a MPT is a commitment to the contents of the tree, which allows a proof to be
+constructed for any key-value mapping encoded in the tree. Such a proof is
+called a Merkle proof, and can be verified against the Merkle root.
+
+## Chain Re-Organization
+[reorg]: /glossary.md#chain-re-organization
+
+A re-organization, or re-org for short, is whenever the head of a blockchain
+(its last block) changes (as dictated by the [fork choice rule]) to a block that
+is not a child of the previous head.
+
+L1 re-orgs can happen because of network conditions or attacks. L2 re-orgs are a
+consequence of L1 re-orgs, mediated via [L2 chain derivation][derivation].
+
+## Precompiled Contract ("Precompile")
+[precompile]: /glossary.md#precompiled-contract-precompile
+
+A contract that exists on chain but consumes no storage, and whose invocation
+does not need to result in the execution of EVM bytecode. Instead, its behaviour
+is defined by specification.
+
+Ethereum L1 precompiles are specified in Appendix E of the [yellow
+paper][yellow].
+
+Optimistic Ethereum exposes all of Ethereum's precompiles, and adds the
+following precompiles:
+
+- [L1 Attributes Precompiled Contract][l1-attr-precompile]
+
+--------------------------------------------------------------------------------
+
+# L2 Chain Concepts
+
+## L2 Chain Inception
+[L2 chain inception]: /glossary.md#L2-chain-inception
+
+The L1 block number for which the first block of the L2 chain was generated.
+
+## Rollup Node
+[rollup node]: /glossary.md#rollup-node
+
+The rollup node is responsible for [deriving the L2 chain][derivation] from the
+[L2 derivation inputs][deriv-inputs] available on L1. This is done by its
+[rollup driver] component.
+
+The rollup node is also responsible for [block gossip], the process of
+transmitting the L2 blocks it derives over a peer-to-peer network.
+
+- cf. [Rollup Node Specification](/rollup-node.md)
+- cf. [Block Gossip Specification](TODO)
+
+> **TODO LINK** block gossip spec
+
+## Rollup Driver
+[rollup driver]: /glossary.md#rollup-driver
+
+The rollup driver is the [rollup node] component responsible for [deriving the
+L2 chain][derivation] from the [L2 derivation inputs][deriv-inputs] available on
+L1.
+
+## L2 Chain Derivation
+[derivation]:  /glossary.md#L2-chain-derivation
+
+A process that reads [L2 derivation inputs][deriv-inputs] from L1 in order to
+derive the L2 chain.
+
+cf. [L2 Chain Derivation (in Rollup Node
+Specification)](/rollup-node.md#l2-chain-derivation)
+
+## L2 Derivation Inputs
+[deriv-inputs]: /glossary.md#l2-chain-derivation-inputs
+
+This term refers to data that is found in L1 blocks and is read by the [rollup
+node] to construct [payload attributes].
+
+Chain derivation attributes include:
+- L1 block attributes
+   - block number
+   - timestamp
+   - basefee
```

Comment:
Perhaps premature optimization, but I could imagine this also having the value of the `RANDOM` opcode post merge, see https://eips.ethereum.org/EIPS/eip-4399

Reply:
How does that relate to the `random` field we need to pass in the [payload attributes](https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#payloadattributesv1)?

Reading through, it seems to be the same:

> Beginning with TRANSITION_BLOCK, client software MUST set the value of the mixHash, i.e. the field with the number 13 (0-indexed) in a block header, to the output of the randomness beacon provided by the beacon chain.

That being said, that only means that it does not mean to be included in the special transaction to convey the attributes... it still needs to be read to construct the attributes, so it might be better to mention it here indeed.

If the `RANDOM` opcode become reality, we'll need to think about wether we want it to return something different on every block. In that case, it might be good to include randomness in the L1 attributes in the transaction, and change the `random` value in the payload attributes.